lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -60,25 +59,33 @@ class JsUsage extends FRGatherer {
    */
   async startSensitiveInstrumentation(context) {
     const session = context.driver.defaultSession;
+    session.on('Debugger.scriptParsed', this.onScriptParsed);
     await session.sendCommand('Debugger.enable');
-    await session.on('Debugger.scriptParsed', this.onScriptParsed);
   }
 
   /**
    * @param {LH.Gatherer.FRTransitionalContext} context
    */
   async stopSensitiveInstrumentation(context) {
     const session = context.driver.defaultSession;
-    await session.off('Debugger.scriptParsed', this.onScriptParsed);
     await session.sendCommand('Debugger.disable');
+    session.off('Debugger.scriptParsed', this.onScriptParsed);
   }
 
   /**
+   * @param {LH.Gatherer.FRTransitionalContext} context
    * @return {Promise<LH.Artifacts['JsUsage']>}
    */
-  async getArtifact() {
+  async getArtifact(context) {
     /** @type {Record<string, Array<LH.Crdp.Profiler.ScriptCoverage>>} */
     const usageByUrl = {};
+
+    // Return url to scriptId mappings in snapshot mode.
```

Comment:
we're not returning in this block, so WDYT about...?

```suggestion
    // Force `Debugger.scriptParsed` events for url to scriptId mappings in snapshot mode.
```